verilog/parser/verilog_token.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/parser/verilog_token.h"
+
+#include <string>
+
+#include "verilog/parser/verilog_parser.h"
+#include "absl/strings/string_view.h"
+
+namespace verilog {
+
+// Strings returned by this function are used as token tags
+// in verible-verilog-syntax' JSON output. Changing them might
+// break third-party code.
+
+std::string TokenTypeToString(verilog_tokentype tokentype) {
+  switch (tokentype) {
+    // Returns stringified symbol name
+    #define CASE_STRINGIFY(val) \
+        case verilog_tokentype::val: return #val;
+
+    // Tokens with verbose aliases
+    CASE_STRINGIFY(TK_COMMENT_BLOCK)
+    CASE_STRINGIFY(TK_EOL_COMMENT)
+    CASE_STRINGIFY(TK_SPACE)
+    CASE_STRINGIFY(TK_NEWLINE)
+    CASE_STRINGIFY(TK_LINE_CONT)
+    CASE_STRINGIFY(TK_ATTRIBUTE)
+    CASE_STRINGIFY(TK_FILEPATH)
+
+    #undef CASE_STRINGIFY
+
+    // Returns token type name or its alias (if available) as used in verilog.y
+    default: {
+      absl::string_view symbol_name(verilog_symbol_name(tokentype));
+      if (symbol_name[0] == '"' || symbol_name[0] == '\'') {
```

Comment:
What should be done about the ASCII quote single-characters?  Should those be protected?  `'` is a token.  Will the `substr` do something weird on "'"?

Reply:
All strings returned by `verilog_symbol_name` for single character tokens are stringified C character literals, so the string representation of `'` character token is `'\''` (i.e. `"'\\''"` in C++ code).
I strip the quotes in order to write e.g `find({"tag": "module"})` instead of `find({"tag": '"module"'})` in Python.

I didn't notice that `\` in quote's string representation - I'll remove it too in order to reference it as `"'"` instead of `"\\'"` in Python.

Reply:
done